### PR TITLE
cleanup rule e2e test

### PR DESCRIPTION
**Pull request by wackxu**
Signed-off-by: wackxu <xushiwei5@huawei.com>

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: tests/e2e/utils/rule.go [hunk 1]
```diff
@@ -204,6 +203,35 @@ func newServiceBusRuleEndpoint() *rulesv1.RuleEndpoint {
...
+	if !modelExists {
+		return errors.New("the requested rule is not found")
+	}
+
+	return nil
+}
+
 // GetRuleList to get the rule list and verify whether the contents of the rule matches with what is expected
 func GetRuleList(list *rulesv1.RuleList, getRuleAPI string, expectedRule *rulesv1.Rule) ([]rulesv1.Rule, error) {
 	resp, err := SendHTTPRequest(http.MethodGet, getRuleAPI)
```
Comment (gy95) [inline tests/e2e/utils/rule.go:237]: delete those not used codes?

Comment (wackxu) [reply tests/e2e/utils/rule.go:237]: yes, will delete it

Comment (wackxu) [reply tests/e2e/utils/rule.go]: done

File: tests/e2e/utils/rule.go [hunk 2]
```diff
@@ -204,155 +202,77 @@ func newServiceBusRuleEndpoint() *rulesv1.RuleEndpoint {
...
-					return nil, errors.New("the rule is not matching with what was expected")
-				}
+
+	return rules.Items, nil
+}
+
+func CheckRuleExists(rules []rulesv1.Rule, expectedRule *rulesv1.Rule) error {
+	modelExists := false
+	for _, rule := range rules {
+		if expectedRule.ObjectMeta.Name == rule.ObjectMeta.Name {
```
Comment (vincentgoat) [inline tests/e2e/utils/rule.go:217]: To reduce code complex, suggestion: 
```
if expectedRule.ObjectMeta.Name != rule.ObjectMeta.Name {
    continue
}
```

File: tests/e2e/deployment/rule_crd_test.go [hunk 3]
```diff
@@ -9,37 +9,35 @@ import (
...
-	RuleHandler         = "/apis/rules.kubeedge.io/v1/namespaces/default/rules"
-)
-
 var _ = Describe("Rule Management test in E2E scenario", func() {
 	var testTimer *utils.TestTimer
 	var testSpecReport SpecReport
+	var edgeClientSet edgeclientset.Interface
+
+	BeforeEach(func() {
+		edgeClientSet = utils.NewKubeEdegClient(ctx.Cfg.KubeConfigPath)
```
Comment (gy95) [inline tests/e2e/deployment/rule_crd_test.go:22]: s/NewKubeEdegClient/NewKubeEdgeClient

File: tests/e2e/utils/rule.go (hunk 2 above)
Comment (wackxu) [reply tests/e2e/utils/rule.go]: done

File: tests/e2e/deployment/rule_crd_test.go (hunk 3 above)
Comment (wackxu) [reply tests/e2e/deployment/rule_crd_test.go]: done

**Pull request merged**